### PR TITLE
Change job application form on /careers/<department>

### PR DESCRIPTION
**Pull request by solazio**
## Done
- Change job application form on `/careers/<department>`

## QA
- Run the website locally using `./run`
- Go to [/careers/sales](http://0.0.0.0:8002/careers/sales) and make sure you can make an application.

## Issue
Fixes https://github.com/canonical-web-and-design/web-squad/issues/1836

### CODE REVIEW

File: static/js/apply-for-jobs.js [hunk 1]
```diff
@@ -107,47 +111,51 @@
...
-  });
+  }
+
+  if (addEducationButton) {
+    addEducationButton.addEventListener("click", () => {
+      numberOfEducations++;
+      addEducationInput(numberOfEducations);
+      // Add click event listener to the "remove education" button
+      const removeEducationButtons = document.querySelectorAll(".js-remove-education");
+      removeEducationButtons[removeEducationButtons.length - 1].addEventListener("click", (e) => {
```
Comment (anthonydillon) [inline static/js/apply-for-jobs.js]: removeEducationButtons maybe empty?

File: static/js/apply-for-jobs.js [hunk 2]
```diff
@@ -107,47 +111,51 @@
...
+        e.preventDefault();
+        const educationToBeRemoved = document.querySelector(`[data-education='${e.target.parentElement.parentElement.dataset.education}']`);
```
Comment (anthonydillon) [inline static/js/apply-for-jobs.js]: This is very reliant on the DOM staying the same. Can you use `closest` to negate that? https://developer.mozilla.org/en-US/docs/Web/API/Element/closest

File: templates/careers/job-detail.html [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      background-size: 80% 100%, 70% 80%, 110% 20%, 100% 100%;
+      padding-bottom: 6rem;
+      padding-top: 6rem;
+    }
+  </style>
+{% endblock %}
+
+{% block overview %}
+  {% if message %}
+  <div class="row">
```
Comment (anthonydillon) [inline templates/careers/job-detail.html]: Missing a p-strip?

File: templates/careers/job-detail.html [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  </div>
+  <div class="row u-sv2">
+      <hr />
+  </div>
+  <div class="row u-extra-space js-roles-apply--container">
+    <div class="col-8">
+      <div class="p-card--strip u-no-margin--bottom">
+        <h3>Apply for this role</h3>
+        <form action="{{ request.path }}" method="POST" enctype="multipart/form-data" class="js-roles-list--form">
+          <fieldset style="border: none; margin-bottom: 0;">
```
Comment (anthonydillon) [inline templates/careers/job-detail.html]: Can you submit a Vanilla propose for this change? I see this inline style a lot on fieldsets

File: webapp/greenhouse_api.py [hunk 5]
```diff
@@ -90,30 +90,43 @@ def get_vacancy(job_id):
     return job
 
 
-def submit_to_greenhouse(form_data, form_cv, job_id="1383152"):
+def submit_to_greenhouse(form_data, form_files, job_id="1383152"):
+    print(form_data)
```
Comment (anthonydillon) [inline webapp/greenhouse_api.py]: Left a print in

File: webapp/greenhouse_api.py [hunk 6]
```diff
@@ -90,30 +90,43 @@ def get_vacancy(job_id):
     return job
 
 
-def submit_to_greenhouse(form_data, form_cv, job_id="1383152"):
+def submit_to_greenhouse(form_data, form_files, job_id="1383152"):
```
Comment (anthonydillon) [inline webapp/greenhouse_api.py]: Can you add a comment about the default job_id. I assume its the ghost one?

File: templates/careers/job-detail.html (hunk 3 above)
Comment (solazio) [reply templates/careers/job-detail.html]: I don't think I need a p-strip here

File: templates/careers/job-detail.html (hunk 4 above)
Comment (solazio) [reply templates/careers/job-detail.html]: done

**Comment by anthonydillon** [comment]
## QA issues:
- When submitting a job application I get a `KeyError: 'GREENHOUSE_API_KEY'`
- On a job details page the Back to list link contains None
- The links to the details are wrong from the department list pages

**Comment by solazio** [comment]
> ## QA issues:
> * When submitting a job application I get a `KeyError: 'GREENHOUSE_API_KEY'`
> * On a job details page the Back to list link contains None
> * The links to the details are wrong from the department list pages

The `GREENHOUSE_API_KEY` is set as environment variable. That's why you get the error.
The back button uses {{ request.referrer }} to create the link, therefore if you land straight on that page, there is no back link
I fixed the link details

**Comment by anthonydillon** [comment]
> The back button uses {{ request.referrer }} to create the link, therefore if you land straight on that page, there is no back link

You should not default to None just empty string

**Comment by solazio** [comment]
> > The back button uses {{ request.referrer }} to create the link, therefore if you land straight on that page, there is no back link
> 
> You should not default to None just empty string

@anasereijo added `/careers/all` as default

**Pull request merged**